src/ReactiveMarbles.Extensions/ReactiveExtensions.cs

```diff
@@ -904,6 +905,91 @@ public static IObservable<Unit> Using<T>(this T obj, Action<T> action, ISchedule
             observer.OnError,
             observer.OnCompleted));
 
+    /// <summary>
+    /// Synchronizes the asynchronous operations in downstream operations.
+    /// Use SubscribeSynchronus instead for a simpler version.
+    /// Call Sync.Dispose() to release the lock in the downstream methods.
+    /// </summary>
+    /// <typeparam name="T">The type of the elements in the source sequence.</typeparam>
+    /// <param name="source">The source.</param>
+    /// <returns>An Observable of T and a release mechanism.</returns>
+    [SuppressMessage("Roslynator", "RCS1047:Non-asynchronous method name should not end with 'Async'.", Justification = "To avoid naming conflicts.")]
```

Comment:
RCS1047 probably doesn't make sense in this project I guess. We can probably turn it off in the .editorconfig